Build failure: python313Packages.jupyter-sphinx

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

No special steps needed.

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/322183018

### Relevant log output

```console
E       sphinx.deprecation.RemovedInSphinx11Warning: 'sphinx.environment.BuildEnvironment.app' is deprecated. Check CHANGES for Sphinx API modifications.

/nix/store/pcj7wxdrljmjjqjqv84hpiqiclb9q1pj-python3.13-sphinx-9.1.0/lib/python3.13/site-packages/sphinx/deprecation.py:89: RemovedInSphinx11Warning
```

### Additional context

Caused by Sphinx 9 upgrade (b15b8d49f63e11078336285d277ceb94cbaba73f). Upstream issue at https://github.com/jupyter/jupyter-sphinx/issues/287.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.19.2, NixOS, 26.05 (Yarara), 26.05.20260217.0182a36`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.3`
 - channels(root): `""`
 - nixpkgs: `/nix/store/dx2qikyb4dyb6hbdfywbmsyla0z5a1h3-source`

### Notify maintainers

[Unmaintained package. I will take care of this later, since it breaks Sage.]


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/sa/sage/sage-src.nix
+++ b/pkgs/by-name/sa/sage/sage-src.nix
@@ -96,6 +96,41 @@ stdenv.mkDerivation rec {
       url = "https://github.com/sagemath/sage/commit/9bd5304f1a222e215d7006a04854ff66616748bf.patch?full_index=1";
       hash = "sha256-7bgouXV3pM20IX5PM24ZpJWOBlZjQksGtjopgouEbyg=";
     })
+
+    # https://github.com/sagemath/sage/pull/41078, landed in 10.8.beta8
+    (fetchpatch2 {
+      name = "docutils-0.22-update.patch";
+      url = "https://github.com/sagemath/sage/commit/e206e205a6841cc5251dfb37fdd36ed29345fba4.patch?full_index=1";
+      hash = "sha256-Y8DqwGBkRnL+6ejZibCmkEJ7q/Qs0wD2KGmAefVdd94=";
+    })
+
+    # https://github.com/sagemath/sage/pull/41342, landed in 10.9.beta1
+    (fetchpatch2 {
+      name = "numpy-2.4-update.patch";
+      url = "https://github.com/sagemath/sage/commit/97ceea842a39fa2f2e9098daae2738a2f2765b9e.patch?full_index=1";
+      hash = "sha256-/Tk3tIy0syOjaNRMCyot6kma3jj4288QJ3zypS79jZo=";
+    })
+
+    # https://github.com/sagemath/sage/pull/41346, landed in 10.9.beta2
+    (fetchpatch2 {
+      name = "ipython-9.8-updte.patch";
+      url = "https://github.com/sagemath/sage/commit/380949e6eeda80cca6e5dd971e2c6f367647a863.patch?full_index=1";
+      hash = "sha256-e4zmgfHrenOixgbUS1uFHzftmwNGGoSb7yFhYmqT0yc=";
+    })
+
+    # https://github.com/sagemath/sage/pull/41395, landed in 10.9.beta2
+    (fetchpatch2 {
+      name = "pyparsing-3.3-update.patch";
+      url = "https://github.com/sagemath/sage/commit/1b5a5dd78b5dcb4b3c7d1f94461ddbc647c5679f.patch?full_index=1";
+      hash = "sha256-kFXg2O3IpwaOwAaNIy6Wscx8/XSDWXrqAXS2ZJgdqsg=";
+    })
+
+    # https://github.com/sagemath/sage/pull/41433, landed in 10.9.beta3
+    (fetchpatch2 {
+      name = "scipy-1.17-update.patch";
+      url = "https://github.com/sagemath/sage/commit/ff58afe27c80c067a8965e1d70966e25d0355aaf.patch?full_index=1";
+      hash = "sha256-gWoXwhUVXL2RSVLPRbxtlP0LCSEkJ9z1PJ1wKLXef1k=";
+    })
   ];
 
   patches = nixPatches ++ bugfixPatches ++ packageUpgradePatches;

--- a/pkgs/by-name/sa/sage/sagelib.nix
+++ b/pkgs/by-name/sa/sage/sagelib.nix
@@ -106,7 +106,7 @@ buildPythonPackage rec {
   ];
 
   pythonRelaxDeps = [
-    "networkx"
+    "sphinx"
   ];
 
   buildInputs = [

--- a/pkgs/development/python-modules/jupyter-sphinx/default.nix
+++ b/pkgs/development/python-modules/jupyter-sphinx/default.nix
@@ -42,8 +42,13 @@ buildPythonPackage rec {
   nativeCheckInputs = [ pytestCheckHook ];
 
   disabledTests = [
-    # https://github.com/jupyter/jupyter-sphinx/issues/280"
+    # https://github.com/jupyter/jupyter-sphinx/issues/280
     "test_builder_priority"
+  ];
+
+  pytestFlags = [
+    # https://github.com/jupyter/jupyter-sphinx/issues/287
+    "-Wignore::sphinx.deprecation.RemovedInSphinx11Warning"
   ];
 
   preCheck = ''

--- a/pkgs/development/python-modules/rpy2-robjects/default.nix
+++ b/pkgs/development/python-modules/rpy2-robjects/default.nix
@@ -53,6 +53,11 @@ buildPythonPackage rec {
 
   nativeCheckInputs = [ pytestCheckHook ];
 
+  pytestFlags = [
+    # https://github.com/rpy2/rpy2/issues/1218
+    "-Wignore::pytest.PytestRemovedIn9Warning"
+  ];
+
   meta = {
     homepage = "https://rpy2.github.io/";
     description = "Python interface to R";

